src/to_ascii.cpp

```diff
@@ -22,7 +22,7 @@ bool constexpr begins_with(std::string_view view, std::string_view prefix) {
   if (view.size() < prefix.size()) {
     return false;
   }
-  return view.substr(0, prefix.size()) == prefix;
+  return view.find(prefix) == 0;
```

Comment:
This change is almost certainly a massive downgrade as far as performance goes. Doing `return view.substr(0, prefix.size()) == prefix;` is just a single `memcmp` call, it is very cheap.

Looking for the position of the string prefix, and then checking that it is zero is conceptually more expensive. Looking at the assembly output in GCC, it is almost much worse.

Unless you have benchmarks supporting this change, I would oppose it, out of caution.

Reply:
I'll do the bench!!